js/treetest.js

```diff
@@ -99,22 +103,38 @@ var tasks = {
 
 	},
 	prev:function() {
-	    if (this.idx === 0) {
-    		this.idx = this.list.length;
-	    }
-	    this.idx = this.idx - 1;
-	    this.set(this.idx);
+		this.recordTimeSpent();
+		if (this.idx === 0) {
+			this.idx = this.list.length;
+		}
+		this.idx = this.idx - 1;
+		this.set(this.idx);
+		this.resetTimer();
 	},
 	set:function(number){
 		this.idx = number;
 		$('#taskDesc').html(this.list[number]);
 		$('#taskNum').html("Task "+(number+1)+" of "+this.list.length);
 		resetTree();
-		if(this.answers[this.idx].length > 0){
-			//need to pass a copy of node path to expandToNode (or else it alters tasks.answers)
-			var copyOfHistory = $.extend(true, [], this.answers[this.idx]);
+		if(this.results[this.idx].answer.length > 0){
+			//need to pass a copy of node path to expandToNode (or else it alters tasks.results[].answer)
+			var copyOfHistory = $.extend(true, [], this.results[this.idx].answer);
 			expandToNode(copyOfHistory);
 		}
+	},
+	resetTimer:function(){
+		this.timerStart = Date.now();
+	},
+	recordTimeSpent:function(){
+		const timerEnd = Date.now();
+		const timerStart = this.timerStart;
+
+		// Otherwise, calling this function twice in a row will double the recorded time spent
+		this.resetTimer();
+
+		// Total time will be added up, since the user could have started on this
+		// task, moved to the next one, then come back to this one
+		this.results[this.idx].timeSpent += timerEnd - timerStart;
```

Comment:
It might be less error prone to record the time on page opening, unless it is already set, and to record the time on submission.
I rather compute the delta when required from raw data possibly.